### PR TITLE
seed RNG before generating color values

### PR DESCRIPTION
**Pull request by tycho**
I was having an issue where the first image I generate always rendered as solid black. It turns out the RNG's seed was being initialized -after- the colors were generated using the RNG, which caused it to generate the same black color values every time.

I've also re-indented everything, but that's optional of course.

### CODE REVIEW

**Comment by debris** [comment]
@alexvandesande merging this pr is very much needed

### ethereum-blockies `0.1.1`
screen shot 2017-07-11 at 11 10 57

### expected
screen shot 2017-07-11 at 11 11 25

**Comment by petejkim** [comment]
I noticed this as well.. also in the sampler, a lot of the seeds are suffixed with a space character, which result in totally different images getting generated.

**Comment by petejkim** [comment]
This is a pretty serious bug, seed of the previous image is being used to generate colors for the next image, so the colors end up being different every time.

Also, do we know what algorithm Etherscan is using?

**Comment by tycho** [comment]
@petejkim Etherscan appears to be doing it right:

```js
    function createIcon(opts) {
        opts = opts || {};
        var size = opts.size || 8;
        var scale = opts.scale || 4;
        var seed = opts.seed || Math.floor((Math.random() * Math.pow(10, 16))).toString(16);
        seedrand(seed);
        var color = opts.color || createColor();
        var bgcolor = opts.bgcolor || createColor();
        var spotcolor = opts.spotcolor || createColor();
        var imageData = createImageData(size);
        var canvas = createCanvas(imageData, color, scale, bgcolor, spotcolor);
        return canvas;
    }
```
(From https://etherscan.io/jss/blockies.js after beautifying.)

**Comment by petejkim** [comment]
Not sure why Etherscan seems to be generating a different image though. Colors are the same, but the image is different:

This is what I get for the address `0x00d1f07af5501156a3dc81ed93f9eebd81d3e472`

screen shot 2017-07-28 at 12 04 58 am

[Etherscan](https://etherscan.io/address/0x00d1f07af5501156a3dc81ed93f9eebd81d3e472):

download

**Comment by balboah** [comment]
If only I had looked at current issues before doing the same change myself

**Comment by alexvandesande** [comment]
Thanks for this. It seems that this fix doesn't fix it all the way, as while the colors are correct, I'm still getting wrong identicons. I will investigate it further.

File: blockies.js [hunk 1]
```diff
@@ -62,23 +62,24 @@
...
-
-    return newOpts;
-  }
+		return newOpts;
+	}
 
-  function renderIcon(opts, canvas) {
-    var opts = buildOpts(opts || {});
+	function renderIcon(opts, canvas) {
+		var opts = buildOpts(opts || {});
```
Comment (alexvandesande) [inline blockies.js]: Found the error. This line should be removed as it restarts the seed again

**Pull request merged**